### PR TITLE
Check keys length before using enacl

### PR DESCRIPTION
**Pull request by velzevur**
[Pivotal story](https://www.pivotaltracker.com/story/show/157389930)

Guard the `enacl` nif from keys with unexpected sizes.
Added some tests.

### CODE REVIEW

File: apps/aetx/src/aetx_sign.erl [hunk 1]
```diff
@@ -56,7 +59,8 @@ sign(Tx, PrivKey) when is_binary(PrivKey) ->
   sign(Tx, [PrivKey]);
 sign(Tx, PrivKeys) when is_list(PrivKeys) ->
     Bin = aetx:serialize_to_binary(Tx),
-    Signatures = [ enacl:sign_detached(Bin, PrivKey) || PrivKey <- PrivKeys ],
+    Signatures = [ enacl:sign_detached(Bin, PrivKey)
+                   || PrivKey <- PrivKeys, ?VALID_PRIVK(PrivKey)],
```
Comment (lucafavatella) [inline apps/aetx/src/aetx_sign.erl]: This is silently filtering. May I suggest an explicit lists:filter if this is intention?

Comment (hanssv) [reply apps/aetx/src/aetx_sign.erl]: I doubt you want to filter here, rather fail in a controlled way?

**Comment by velzevur** [comment]
The main point was not crashing when verifying incoming keys (because it could be an attack vector)

What relates to a node signing with a broken key - this probably is a symptom for a bug so modified the silent producing of broken _signed_ transactions into a loud bang.

**Comment by lucafavatella** [comment]
I see comment clarified. Removing WIP label.

**Comment by velzevur** [comment]
Addressed comments:

* coinbases transactions in tests are spend transactions so one less dependency in removing the coinbase

* error emitted by `aetx_sign:sign2` does contain the invalid key(s)

* `aetx_sign:verify_one_pubkey/4` head that was processing 2 errors, now is split into two distinct heads for better visibility

**Pull request merged**